native.go

```diff
@@ -209,30 +209,26 @@ func (r *Runtime) CreateNativeFunction(name, file string, call func(FunctionCall
 func (r *Runtime) Eval(name, src string, direct, strict bool) (Value, error) {
 	this := r.NewObject()
 
-	p, err := r.compile(name, src, strict, true)
+	vm := r.vm
+	p, err := r.compile("<eval>", src, strict, true, !direct || vm.stash == &r.global.stash)
```

Comment:
I think `r.compile` should take `name` as first param and not `"<eval>"`, this was coming from `eval` but wanted to double check that we should use the `name` param.